src/main/java/io/hops/hopsworks/expat/migrations/featurestore/statistics/StatisticsMigration.java

```diff
@@ -772,39 +798,49 @@ private String statisticsFileName(Long startCommitTime, Long endCommitTime, Stri
     return name + endCommitTime + "_" + featureName + ".json";
   }
   
-  private boolean updateFeatureGroupStatisticsCommitWindow(PreparedStatement updateFgsStmt, Integer fgStatisticsId,
-    Long windowStartCommitTime) throws SQLException {
+  private boolean loadFeatureGroupStatisticsCommitWindow(Set<FeatureGroupStatisticsCommitWindow> loadList,
+      Integer fgStatisticsId, Long windowStartCommitTime) throws SQLException {
     if (windowStartCommitTime == null) {
       return false; // skip update, no time travel-enabled feature group
     }
-    updateFgsStmt.setLong(1, windowStartCommitTime);
-    updateFgsStmt.setInt(2, fgStatisticsId);
-    updateFgsStmt.addBatch();
+    loadList.add(new FeatureGroupStatisticsCommitWindow(fgStatisticsId, windowStartCommitTime));
     return true;
   }
-  
-  private void deleteFeatureGroupStatistics(PreparedStatement deleteFgsStmt, Integer fgStatisticsId)
-      throws SQLException {
-    deleteFgsStmt.setInt(1, fgStatisticsId);
-    deleteFgsStmt.addBatch();
-  }
-  
-  private void deleteFeatureGroupStatisticsActivity(PreparedStatement deleteFgsActStmt, Integer fgStatisticsId)
-    throws SQLException {
-    deleteFgsActStmt.setInt(1, fgStatisticsId);
-    deleteFgsActStmt.addBatch();
+
+  private void updateFeatureGroupStatisticsCommitWindow(PreparedStatement updateFgsStmt,
+      Set<FeatureGroupStatisticsCommitWindow> listToUpdate) throws SQLException {
+    int c = 0;
+    for (FeatureGroupStatisticsCommitWindow e : listToUpdate) {
+      updateFgsStmt.setLong(1, e.windowStartCommitTime);
+      updateFgsStmt.setInt(2, e.fgStatisticsId);
+      updateFgsStmt.addBatch();
+      if (c % statisticsMigrationBatchSize == 0) {
+        LOGGER.info(String.format("[migrateFeatureDescriptiveStatistics] Update FGS: %s", updateFgsStmt.toString()));
+        updateFgsStmt.executeBatch();
+      }
+      c++;
+    }
+    LOGGER.info(String.format("[migrateFeatureDescriptiveStatistics] Update FGS: %s", updateFgsStmt.toString()));
+    updateFgsStmt.executeBatch();
   }
   
-  private void deleteTrainingDatasetStatistics(PreparedStatement deleteTdsStmt, Integer tdStatisticsId)
+  private void deleteStatisticsBatch(PreparedStatement deleteStatisticsStmt, Set<Integer> statisticsIdsToDelete,
+      String log)
       throws SQLException {
-    deleteTdsStmt.setInt(1, tdStatisticsId);
-    deleteTdsStmt.addBatch();
-  }
-  
-  private void deleteTrainingDatasetStatisticsActivity(PreparedStatement deleteTdsActStmt, Integer tdStatisticsId)
-    throws SQLException {
-    deleteTdsActStmt.setInt(1, tdStatisticsId);
-    deleteTdsActStmt.addBatch();
+    int c = 0;
```

Comment:
```suggestion
    int c = 1;
```
Same comment as for update sql queries.